Automattic/wp-calypso
# Calypso CSS: refactor CSS reset and base styles

Goal: scope CSS reset and base calypso styles so that it's possible to exclude them from `/gutenboarding` section.

Specifically, I'm addressing these two issues from https://github.com/Automattic/wp-calypso/pull/36170#issuecomment-540571341:
- CSS reset stylesheet
- styles for form elements, as we apply Calypso styling even to plain input or select elements that don’t have any class names

Here are [some Calypso overrides](https://github.com/Automattic/wp-calypso/blob/17380909031ea785fcabd63e51562f5b05f0bdb2/client/gutenberg/editor/style.scss#L296-L407) previous "Gutenberg in Calypso" had to do.

#### Changes proposed in this Pull Request

* Move scss partials to `client/assets/stylesheets/layout`, leaving `client/assets/stylesheets/*.scss` only for entry files
* Move animations partial to new `layout` folder, leaving shared for only scss files that don't produce any visible CSS when imported
* Move `grow` animation to `_animations.scss`
* Change loading order so that resets and base styles come first
* Move Calypso styles that affect non-class selectors to a new `_base.scss` file, goal is to strip down generic "main" file away eventually
* Scope reset and base -styles so that it's possible to exclude them from some views. TODO: an exact mechanism to be figured out. I'm using `html.gr__calypso_localhost` as a temporary example selector — it'll be something else later on.
* Update outdated `README.md` 

I'll PR some of this separately for easier review & testing:
- https://github.com/Automattic/wp-calypso/pull/37026
- https://github.com/Automattic/wp-calypso/pull/37064

#### Testing instructions

* Spin up Calypso
* Observe that base and reset styles are _first_ in entry bundle
* Test everything!
* `/onboarding` looks great, eh?
